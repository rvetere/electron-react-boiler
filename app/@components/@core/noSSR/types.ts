import { ReactNode } from 'react'

export interface INoSSRProps {
  onSSR?: ReactNode
  renderOnServerForBots?: boolean
}

export interface INoSSRState {
  mounted: boolean
}
