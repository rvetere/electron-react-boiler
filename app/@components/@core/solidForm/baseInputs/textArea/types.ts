import { IValidationDisplayConfig } from '@core/solidForm/elements/infoSlot/validationDisplay/types'
import { IFormControllerInputProps } from '@core/solidForm/formController/types'
import { IInputDefaultState } from '@core/solidForm/types'
import { ValidatorFn } from '@core/solidForm/validators/types'
import React, { ReactNode } from 'react'

export interface ITextAreaProps extends React.InputHTMLAttributes<HTMLTextAreaElement> {
  name: string
  inputProps: IFormControllerInputProps
  validators?: ValidatorFn[]
  inverted?: boolean
  label?: ReactNode
  labelAppendix?: ReactNode
  hideValidation?: boolean
  infoSlot?: (validationConfig: IValidationDisplayConfig) => ReactNode
}

export type ITextAreaState = IInputDefaultState
