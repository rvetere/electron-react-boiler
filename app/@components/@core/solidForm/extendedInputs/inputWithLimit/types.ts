import { IInputTextProps } from '@core/solidForm/baseInputs/input/types'
import { ITextAreaProps } from '@core/solidForm/baseInputs/textArea/types'
import { FieldWithLimitType } from '.'

export interface IInputTextWithLimitProps extends IInputTextProps {
  limit: number
  type?: FieldWithLimitType
}

export interface ITextAreaWithLimitProps extends ITextAreaProps {
  limit: number
  type?: FieldWithLimitType
}

export interface IInputWithLimitState {
  isFocused: boolean
  valueLength: number
}
