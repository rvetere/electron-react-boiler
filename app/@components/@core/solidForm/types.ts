import { InputValue } from '.'
import { IValidationMessage } from './elements/infoSlot/validationDisplay/types'

export enum LabelPosition {
  LabelLeft = 'LabelLeft',
  LabelRight = 'LabelRight'
}

export interface IInputDefaultState {
  value?: InputValue
  isValid: boolean
  isTouched: boolean
  validationMessages: IValidationMessage[]
}
