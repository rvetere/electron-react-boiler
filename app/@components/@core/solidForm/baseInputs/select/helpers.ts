import { ISelectOption } from './types'

export const getSelectedOption = (givenValue: string | number | null, options: ISelectOption[]): ISelectOption | undefined =>
  options.find(({ value }: ISelectOption): boolean => value === givenValue)
