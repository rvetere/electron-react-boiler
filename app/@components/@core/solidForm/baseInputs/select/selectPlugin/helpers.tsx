import { ChevronDown, ChevronUp, Close } from '@core/image/icon'
import { ISelectOption } from '../types'

export const ArrowRenderer = ({ isOpen }: { isOpen: boolean }): JSX.Element => {
  if (isOpen) {
    return <ChevronUp />
  }

  return <ChevronDown />
}

export const ClearRenderer = (): JSX.Element => <Close />

export const unClearableOptions = (options?: ISelectOption[]): ISelectOption[] => {
  if (!options) {
    return []
  }

  return options.map(
    (option: ISelectOption): ISelectOption => ({
      ...option,
      clearableValue: false
    })
  )
}
