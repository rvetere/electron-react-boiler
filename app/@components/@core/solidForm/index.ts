export { Checkbox } from './baseInputs/checkbox'
export { ICheckboxProps } from './baseInputs/checkbox/types'
export { InputText } from './baseInputs/input'
export { IInputTextProps } from './baseInputs/input/types'
export { TextArea } from './baseInputs/textArea'
export { ITextAreaProps } from './baseInputs/textArea/types'
export { RadioGroup } from './baseInputs/radioGroup'
export { IRadioGroupProps, IRadioProps } from './baseInputs/radioGroup/types'
export { Select } from './baseInputs/select'
export { getSelectedOption } from './baseInputs/select/helpers'
export { ISelectProps, ISelectOption } from './baseInputs/select/types'
export { IGroupType } from './baseInputs/select/selectPlugin/types'
export { RadioOption } from './baseInputs/radioGroup/radio'
export { IRadioOptionProps } from './baseInputs/radioGroup/radio/types'
export { InputPassword } from './extendedInputs/inputPassword'
export { RadioRow } from './extendedInputs/radioRow'
export { InputWithLimit, FieldWithLimitType } from './extendedInputs/inputWithLimit'
export { IInputTextWithLimitProps, ITextAreaWithLimitProps } from './extendedInputs/inputWithLimit/types'
export { FormController } from './formController'
export {
  IFormControllerProps,
  IFormControllerActions,
  IFormValues,
  FormSubmitFn,
  FormResetFn,
  IFormControllerInputProps,
  IFormControllerFields,
  IFormControllerField
} from './formController/types'
export { LabelPosition } from './types'

export type InputValue = string[] | string | number | undefined
