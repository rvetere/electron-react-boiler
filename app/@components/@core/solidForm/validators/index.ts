import isPostalCode from './helpers/isPostalCode'
import { IValidatorResult } from './types'

export const RequiredMsgDefault = 'Required'

export const longerThan5 = (value: string): IValidatorResult => ({
  isValid: value.length > 5,
  message: { message: 'More than 5 characters' }
})

export const containsUpperCase = (value: string): IValidatorResult => ({
  isValid: /[A-Z]/.test(value),
  message: { message: 'Min 1 UpperCase' }
})

export const isZip = (value: string): IValidatorResult => ({
  isValid: isPostalCode(value),
  message: { message: 'Invalid zip' }
})
