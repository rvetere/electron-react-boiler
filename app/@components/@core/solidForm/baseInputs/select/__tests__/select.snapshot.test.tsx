/* eslint-env jest */
import { getInputProps } from '@core/solidForm/__tests__/__mocks__/solidForm.mock'
import { selectOptions } from '@misc/styleguide/solidForm/sample/helpers'
import { getByClass } from '@testing/classQueries'
import { cleanup, fireEvent, render, waitForElement } from 'react-testing-library'
import { Select } from '..'
import { ISelectProps } from '../types'

describe('Select', () => {
  afterEach(cleanup)
  const inputProps = getInputProps()
  const mockProps: ISelectProps = {
    inputProps,
    name: 'select',
    label: 'Select'
  }

  test('Select component should render correctly', () => {
    const { container } = render(<Select {...mockProps} options={selectOptions} />)

    expect(container).toMatchSnapshot()
    expect(mockProps.inputProps.registerInput).toBeCalledTimes(1)
  })

  test('Select should validate', async () => {
    const { container } = render(<Select {...mockProps} required options={selectOptions} />)

    const elem = container.querySelector('select')
    fireEvent.blur(elem!)
    fireEvent.change(elem!, { target: { value: undefined } })

    const validationDisplay = await waitForElement(() => getByClass(container, 'validationMessage'))

    expect(validationDisplay).toBeTruthy()
    expect(mockProps.inputProps.updateForm).toBeCalledTimes(1)
  })
})
