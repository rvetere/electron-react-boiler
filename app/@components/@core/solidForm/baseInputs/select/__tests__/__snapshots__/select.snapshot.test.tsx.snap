// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Select component should render correctly 1`] = `
<div>
  <div
    class="formField optional"
  >
    <label
      class="formLabel label"
    >
      <span>
        Select
      </span>
      <div
        class="wrapper"
      >
        <select
          aria-describedby="selectValidationMessage"
          class="select"
          id="select"
          name="select"
        >
          <option
            value="0"
          >
            Option 1
          </option>
          <option
            value="1"
          >
            Option 2
          </option>
          <option
            value="2"
          >
            Option 3
          </option>
          <option
            value="3"
          >
            Option 4
          </option>
          <option
            value="4"
          >
            Option 5
          </option>
          <option
            value="5"
          >
            Option 6
          </option>
          <option
            value="6"
          >
            Option 7
          </option>
          <option
            value="7"
          >
            Option 8
          </option>
          <option
            value="8"
          >
            Option 9
          </option>
          <option
            value="9"
          >
            Option 10
          </option>
          <option
            value="10"
          >
            Option 11
          </option>
          <option
            value="11"
          >
            Option 12
          </option>
          <option
            value="12"
          >
            Option 13
          </option>
          <option
            value="13"
          >
            Option 14
          </option>
          <option
            value="14"
          >
            Option 15
          </option>
          <option
            value="15"
          >
            Option 16
          </option>
          <option
            value="16"
          >
            Option 17
          </option>
          <option
            value="17"
          >
            Option 18
          </option>
          <option
            value="18"
          >
            Option 19
          </option>
          <option
            value="19"
          >
            Option 20
          </option>
        </select>
        <test-file-stub
          classname="digicon"
          height="16"
          width="16"
        />
      </div>
    </label>
  </div>
</div>
`;
